src/rhalphalib/sample.py

```diff
@@ -225,7 +225,12 @@ def setParamEffect(self, param: Parameter, effect_up, effect_down=None, scale=No
             effect_up, binning, _ = _to_numpy(effect_up)
             if not np.array_equal(binning, self.observable.binning):
                 raise ValueError("effect_up has incompatible binning with sample %r" % self)
-            zerobins = (self._nominal <= 0.0) | (effect_up <= 0.0)
+            zerobins_nominal = (self._nominal <= 0.0) | np.isclose(self._nominal, 0.0)
+            zerobins_effect = (effect_up <= 0.0) | np.isclose(effect_up, 0.0)
+            _fraction_non_matching = sum(~zerobins_nominal == ~zerobins_effect) / len(zerobins_nominal)
+            if _fraction_non_matching <= 0.85 and not nowarn:
```

Comment:
Why 0.85? Can assign it as a class static member `NonMatchingThreshold` or whatever makes sense with a docstring about why.